src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -348,6 +356,65 @@ public override void WriteEndAttribute()
             }
         }
 
+        protected override async Task WriteEndAttributeAsync()
+        {
+            if (IsClosed)
+                ThrowClosed();
```

Comment:
Here's an example where the exception is being stored into the returned Task (because this method is async).  We should be consistent, both with these additions and with what's already done in the library.